src/dinitctl.cc

```diff
@@ -294,6 +295,10 @@ int dinitctl_main(int argc, char **argv)
         }
         else {
             // service name / other non-option
+            if (argv[i][0] == '\0') {
+                empty_arg_error = true;
```

Comment:
You can just report the error here. (What benefit is there in setting a variable here and testing it later? Look at how other errors are reported (other than the generic case), and be consistent with that).